File: escomvoting-backend/src/main/java/com/github/escom/escomvoting/model/dto/CreateElectionRequest.java
package com.github.escom.escomvoting.model.dto;

import java.time.Instant;
import java.util.List;

public record CreateElectionRequest(
        String title,
        String description,
        Instant startDate,
        Instant endDate,
        List<String> allowedRoles,
        List<CandidateInput> candidates
) {
    public record CandidateInput(String name, String description) {}
}

File: escomvoting/src/model/request/CreateElectionRequest.ts
export interface CandidateInput {
  name: string
  description: string
}

export interface CreateElectionRequest {
  title: string
  description: string
  startDate: string   // ISO-8601 instant, e.g. "2025-06-01T09:00:00Z"
  endDate: string
  allowedRoles: string[]  // "STUDENT" | "PROFESSOR" | "PAAE"
  candidates: CandidateInput[]
}
